### ISSUE TITLE
Mobile specific UI

### ISSUE THREAD

**Issue by lommes** [gogs/gogs#144]
... to make them fit on mobile devices. 

Same goes with class img-responsive for images like on startpage

**Comment by EtienneBruines** [comment]
Also affects non-mobile devices that have enabled zoom (i.e.: pages showing at 110% zoom). 

For instance, the "copy" button at newly created `repo-clone`, is not shown on the same page.

**Comment by Unknwon** [comment]
@EtienneBruines thanks your info!

**Comment by bastiao** [comment]
Also, in the Issues, "New issue" and "Milestones" are not available. Tested in iPhone.

**Comment by nanoant** [comment]
@Unknwon FYI I can help you with that, if you want. As long we stick with pure Semantic UI where possible, this shouldn't be hard. See #2066

**Comment by Unknwon** [comment]
@nanoant thanks for the advice, but Gogs is very rich content app, so I think simple responsive design is not going to work.

However, we need a complete different design page for mobile devices, and for sure it is going to use SUI.

**Comment by nanoant** [comment]
@Unknwon Since we don't have sidebar (which is good), which makes single column main layout, it is just a matter of properly configuring various elements to collapse properly on mobile devices. Somehow the PR of mine #2068 is example that now file list and commit list is somehow readable on mobile device, because layout no longer breaks (eg. icons breaking of the names, etc).

But I am aware there are some other places, such as top menu, where we could collapse to icon only view, removing textual labels such as "Pull request" that will no longer fit, or make it alternatively stackable.

We don't need perfect Mobile UI, but usable one, and I think we can manage it having single HTML+CSS using Stackable UI device (screen) classes.

**Comment by Unknwon** [comment]
@nanoant I think we should the simple design principle of GitHub, which means we do not have to show everything as desktop version does.

**Comment by tboerger** [comment]
I really like the idea to be more mobile friendly. This view doesnt need to be officially supported yet. A more or less usable state until we have a real mobile view should be already fine.

**Comment by linquize** [comment]
What framework can be used to support mobile view?

**Comment by Unknwon** [comment]
@linquize I think SUI is good enough?

**Comment by linquize** [comment]
Is it?  https://github.com/SUI-Components

**Comment by Unknwon** [comment]
I think point of mobile friendly is use more space for less content, does not matter the UI components.

**Comment by khromov** [comment]
Responsive layout would be great! Just looking at the current CSS, there are a few smart tweaks we could do to make it work nicely on a mobile device.

**Comment by YamiOdymel** [comment]
gogs was made by Semantic-UI, it has the built-in  mobile tools.

**Comment by tboerger** [comment]
> gogs was made by Semantic-UI, it has the built-in mobile tools.

Only if it's done right ;)

**Comment by YamiOdymel** [comment]
I was about to ask if there's any chances to send a pull request about this.

Sorry but why isn't it allowed?

**Comment by AddoSolutions** [comment]
I have quite an extensive UI background, if my assistance is of any value. I have done a lot with Semantic UI as well

**Comment by bkcsoft** [comment]
@AddoSolutions I at least want it more mobile-friendly :smiley: ( though I'd prefer an app honestly :trollface: ). If you decide to work on it, try as hard as you can to only do it with CSS, as fscking around with styles inline/javascript is horrible at best :unamused:

@YamiOdymel The reason is that Gogs has some inline-styles, and javascript fscking up styles. So it's not an easy task. Also MobileUI is very low priority as (like I mentioned above) a Mobile App would be preferred (when the API stabilizes I'll have a look at forking ForkHub :smiley: )

**Comment by genofire** [comment]
It should not be so difficult just add
```
<meta name="viewport" content="width=device-width, initial-scale=1, user-scalable=no" />
```
and add on the most element with class `grid` or `menu` the class `stackable`.
(So it look not good on mobile, but is useable -> https://github.com/genofire/gogs/commit/14fcd38a7566b5f3ee483ad6ad04ae67605214ab)

**Comment by rugk** [comment]
Any news on this?
And, @Unknwon, you still have not said why you added the "don't send pull requests" label.

**Comment by Unknwon** [comment]
No, at least not until 2018.

**Comment by rugk** [comment]
And what about this:
> And, @Unknwon, you still have not said why you added the "don't send pull requests" label.

:expressionless:

**Comment by YamiOdymel** [comment]
Because the UI is still changing, it's not stable yet.
2016/8/7 上午12:46 於 "rugk" <notifications@github.com> 寫道：

**Comment by rugk** [comment]
Okay.

**Comment by Unknwon** [comment]
It means we won't accept your PR about this, you can make it but only in your fork. It's a warning.

**Comment by evantbyrne** [comment]
The reasoning behind not accepting PRs on this doesn't make much sense to me. Typically mobile UI dev is done alongside the desktop UI dev, or even before because elements naturally stack vertically in HTML. I would recommend considering PRs on this. There are probably a lot of people in this community willing to improve the mobile layout, including myself. I don't think it would take much to get it usable on mobile.

**Comment by YamiOdymel** [comment]
For me, Gogs is still unstable (changes often), 

I think it's acceptable to tell people "don't send pull request" for the UI (at least now),

it's like a unfinished game, people would like to translate it, ask PR for it, 

but the strings of the game might be changed after a _little update_, sooner or later, 

those translate will be useless, it's something I felt now.

**Comment by vavrecan** [comment]
Its November already. Opensource is about collaboration. I was wondering, making its responsive is just a matter of putting in stackable and few easy changes. I think if the admins will not accept pull requests and does not want other people to contribute to the project (which I believe there are many great programmers), we can just make fork (like lede from openwrt) voila :+1:

**Comment by andreynering** [comment]
This is a 2.5 year old issue. Since there are people willing to work on this, I also think that PRs should be accepted.

**Comment by tboerger** [comment]
It have been labeled that no pull requests are getting accepted... :(

**Comment by andreynering** [comment]
PRs for this are being accepted for [Gitea](https://github.com/go-gitea/gitea) fork.

https://github.com/go-gitea/gitea/issues/35

**Comment by lyuwen** [comment]
I saw this issue... I want to post a new one because it is 3 years old. @Unknwon

**Comment by Unknwon** [comment]
@lyuwen issues are valid as long as it is open no matter how old it is :smiley:

**Comment by skylarmt** [comment]
> No, at least not until 2018

    ~$ date
    Tue Jan 16 00:28:45 MST 2018

**Comment by netlander** [comment]
Arrogant git @Unknwon

**Comment by agherzan** [comment]
Any news on this?

**Comment by Unknwon** [comment]
Unfortunately, no... Looking forward to use VueJS but still learning it.

**Comment by BTaskaya** [comment]
> No, at least not until 2018

its 2019 and we are still waiting

**Comment by skylarmt** [comment]
If you want a mobile interface, consider switching to Gitea, a community-run fork of Gogs.  It has a mobile interface and some other features Gogs doesn't, such as a dark theme and two-factor with hardware security keys.

**Comment by baxandrei** [comment]
Any news on this?

**Comment by skylarmt** [comment]
I've since switched to Gitea, a fork of Gogs but it has a responsive/mobile UI, dark themes, and at least one Android app.

**Comment by ElleshaHackett** [comment]
Please support this. I'm happy to help too. This should prevent some users from moving to Gitea as well. It's 2020. That's simply requirement nowadays. Even adaptive (not fully responsive) support or showing less functionality is a start.

**Comment by unknwon** [comment]
@ElleshaHackett Mobile specific UI is where I plan to use VueJS and GraphQL but my skills on the former is still rough 😅

**Comment by Schlomon** [comment]
Any news on this?

**Comment by skylarmt** [comment]
> Any news on this?

My company has switched over to Gitea, a fork of Gogs that has a lot more community involvement, which has led to improvements such as a responsive UI and two-factor authentication. There's also a third-party Gitea Android app in F-Droid called GitNex.

**Comment by Schlomon** [comment]
> My company has switched over to Gitea, a fork of Gogs that has a lot more community involvement, which has led to improvements such as a responsive UI and two-factor authentication. There's also a third-party Gitea Android app in F-Droid called GitNex.

Yeah I more and more feel like gitea is the only real option here.

**Comment by Prochy20** [comment]
Support 🔝🔝

**Comment by devraza** [comment]
Support ^^
